src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Utils/SPDXExtensions.cs

```diff
@@ -19,158 +19,130 @@ namespace Microsoft.Sbom.Parsers.Spdx30SbomParser.Utils;
 /// </summary>
 public static class SPDXExtensions
 {
-    /// <summary>
-    /// Only these chars are allowed in a SPDX id. Replace all other chars with '-'.
-    /// </summary>
-    private static readonly Regex SpdxIdAllowedCharsRegex = new Regex("[^a-zA-Z0-9.-]");
+    private const string SpdxIdPrefix = "SPDXRef";
 
     /// <summary>
-    /// Returns the SPDX-compliant ID for a general element.
-    /// <paramref name="element"/> The element to generate the ID for.
-    /// <paramref name="id"/> The ID that uniquely identifies an element to generate the hash for.
-    /// </summary>
-    public static string GenerateSpdxId(Element element, string id) => $"SPDXRef-{element.Type}-{GetStringHash(id)}";
-
-    public static string GetSpdxId(Element element, string id)
-    {
-        var spdxFileId = $"{Constants.SPDXRefFile}-{element.Name}-{id}";
-        return SpdxIdAllowedCharsRegex.Replace(spdxFileId, "-");
-    }
-
-    /// <summary>
-    /// Returns the SPDX-compliant external document ID.
+    /// Adds ExternalSpdxId property to the SPDXExternalDocumentReference based on name and checksum information.
     /// </summary>
-    public static string GenerateSpdxExternalDocumentId(string fileName, string sha1Value)
+    public static string AddExternalSpdxId(this ExternalMap reference, string name, IEnumerable<Checksum> checksums)
     {
-        var spdxExternalDocumentId = $"DocumentRef-{fileName}-{sha1Value}";
-        return SpdxIdAllowedCharsRegex.Replace(spdxExternalDocumentId, "-");
-    }
+        if (reference is null)
+        {
+            throw new ArgumentNullException(nameof(reference));
+        }
 
-    /// <summary>
-    /// Get's an ID that corresponds to the package info
-    /// </summary>
-    /// <param name="spdxPackage"></param>
-    /// <param name="packageInfo"></param>
-    /// <returns>Package ID that encapsulates unique info about a package.</returns>
-    public static string GetSpdxElementId(SbomPackage packageInfo)
-    {
-        if (packageInfo is null)
+        if (string.IsNullOrEmpty(name))
         {
-            throw new ArgumentNullException(nameof(packageInfo));
+            throw new ArgumentException($"'{nameof(name)}' cannot be null or empty.", nameof(name));
```

Comment:
ArgumentException will already include the parameter name. This can be simplified to
```suggestion
            throw new ArgumentException("Parameter cannot be null or empty.", nameof(name));
```